### PR TITLE
Remove Ambiguous Behavior of Tuple as Grouping

### PR DESCRIPTION
**Pull request by WillAyd**
follow up to #18314

### CODE REVIEW

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -343,14 +345,25 @@ def _group_selection_context(groupby):
     groupby._reset_group_selection()
 
 
+KeysArgType = Optional[
+    Union[
+        Hashable,
+        List[Hashable],
+        Callable[[Hashable], Hashable],
+        List[Callable[[Hashable], Hashable]],
```
Comment (WillAyd) [inline pandas/core/groupby/groupby.py]: Pretty large signature and maybe worth trimming down / cleaning up (see #22278) but documenting as-is state

Comment (jbrockmendel) [reply pandas/core/groupby/groupby.py]: yikes.  maybe a comment pointing to the relevant issue?

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -343,14 +345,25 @@ def _group_selection_context(groupby):
     groupby._reset_group_selection()
 
 
+KeysArgType = Optional[
```
Comment (simonjayhawkins) [inline pandas/core/groupby/groupby.py]: in general maybe better not to include the Optional in the alias, and just add the Optional where needed when using the alias. I find this promotes better resuse of aliases.

Comment (WillAyd) [reply pandas/core/groupby/groupby.py]: Yea I went back and forth on this myself. What swayed me to include it here is that outside of the signature it is valid for this object to always be `None`, so long as it is paired with a valid `level`. But happy to change

File: doc/source/whatsnew/v1.0.0.rst [hunk 3]
```diff
@@ -313,6 +313,7 @@ or ``matplotlib.Axes.plot``. See :ref:`plotting.formatters` for more.
 - Removed the previously deprecated :meth:`Series.get_value`, :meth:`Series.set_value`, :meth:`DataFrame.get_value`, :meth:`DataFrame.set_value` (:issue:`17739`)
 - Changed the the default value of `inplace` in :meth:`DataFrame.set_index` and :meth:`Series.set_axis`. It now defaults to False (:issue:`27600`)
 - Removed support for nested renaming in :meth:`DataFrame.aggregate`, :meth:`Series.aggregate`, :meth:`DataFrameGroupBy.aggregate`, :meth:`SeriesGroupBy.aggregate`, :meth:`Rolling.aggregate` (:issue:`18529`)
+- A tuple passed to :meth:`DataFrame.groupby` is now exclusively treated as a single key (:issue:`18314`)
```
[the remark this answers is not in the document]
Comment (WillAyd) [reply doc/source/whatsnew/v1.0.0.rst]: We aren't changing the intended behavior just removing deprecated arguments, so I think should belong here

File: pandas/core/groupby/groupby.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/core/groupby/groupby.py]: also I _think_ aliases should always start with an underscore.

File: pandas/tests/groupby/test_groupby.py [hunk 4]
```diff
@@ -1734,34 +1734,23 @@ def test_empty_dataframe_groupby():
...
     # https://github.com/pandas-dev/pandas/issues/18314
     df = pd.DataFrame(
         {
-            ("a", "b"): [1, 1, 2, 2],
-            "a": [1, 1, 1, 2],
-            "b": [1, 2, 2, 2],
+            ("a", "b"): [1, 1, 1, 1],
+            "a": [2, 2, 2, 2],
+            "b": [2, 2, 2, 2],
             "c": [1, 1, 1, 1],
```
Comment (jbrockmendel) [inline pandas/tests/groupby/test_groupby.py]: why does this dataframe need to change?  the rest of the test change makes sense

Comment (WillAyd) [reply pandas/tests/groupby/test_groupby.py]: Didn't need to just figured easier to disambiguate between this and the respective a / b columns if they have entirely different values

File: pandas/core/groupby/groupby.py [hunk 5]
```diff
@@ -343,14 +345,23 @@ def _group_selection_context(groupby):
     groupby._reset_group_selection()
 
 
+_KeysArgType = Union[
```
Comment (jreback) [inline pandas/core/groupby/groupby.py]: this is a monster. can you add some descriptions to it on each line (followon ok); *maybe* this should be in _typing?

**Comment by jreback** [comment]
thanks @WillAyd small followon request.

**Comment by jbrockmendel** [comment]
@WillAyd pls be sure to update #6581/#13777 if called for

**Comment by jbrockmendel** [comment]
@WillAyd looks like you marked the checkbox in #6581; the process is to move the entry to #13777 and add in parens the PR reference.

**Comment by WillAyd** [comment]
My mistake - should be updated now

**Pull request merged**